blue_hardware_drivers/src/BLDCDriver.cpp

```diff
@@ -258,4 +281,24 @@ void BLDCDriver::engageControl() {
   engaged_ = true;
 }
 
+bool BLDCDriver::setControlMode(int id, comm_ctrl_mode_t control_mode){
+  bool success = false;
+  if (board_control_modes[id] != control_mode) {
```

Comment:
consider:
```
void BLDCDriver::setControlMode(int id, comm_ctrl_mode_t control_mode){
  while (board_control_modes[id] != control_mode && ros::ok()) {
    try {
      // stuff
    } catch () {
      // stuff
    }
  }
}
```

(the bool return is currently not useful, and the success variable + if statement could be removed)